neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/utils/BoltNeo4jUtils.java

```diff
@@ -117,4 +120,30 @@ public static void closeSafely(AutoCloseable closeable, Logger logger) {
         }
     }
 
+    public static boolean hasResultSet(BoltNeo4jConnection connection, String cypher) {
+        try (final Session session = connection.newNeo4jSession();
+             final Transaction transaction = session.beginTransaction()) {
+            if (StringUtils.isBlank(cypher)) {
+                return false;
+            }
+            final String toUpperCase = cypher.trim().toUpperCase();
+            final String cql;
+            if (toUpperCase.startsWith("EXPLAIN") || toUpperCase.startsWith("PROFILE")) {
+                cql = cypher;
+            } else {
+                cql = "EXPLAIN " + cypher;
+            }
+            return !transaction
+                    .run(cql)
+                    .consume()
+                    .plan()
+                    .children()
+                    .stream()
+                    .map(f -> f.operatorType())
+                    .anyMatch(o -> o.equalsIgnoreCase("EmptyResult"));
```

Comment:
nit: you can replace `![...].anyMatch` with `noneMatch`, I think

Reply:
done